src/trio/_tests/test_highlevel_socket.py

```diff
@@ -283,7 +289,7 @@ async def accept(self) -> tuple[SocketType, object]:
 
     for code in [errno.EMFILE, errno.EFAULT, errno.ENOBUFS]:
         with assert_checkpoints():
-            with pytest.raises(OSError) as excinfo:
+            with pytest.raises(OSError) as excinfo:  # noqa: PT011  # missing `match`
                 await listener.accept()
             assert excinfo.value.errno == code
```

Comment:
```python
for code, match in zip((errno.EMFILE, errno.EFAULT, errno.ENOBUFS), ("Out of file descriptors", "attempt to write to read-only memory", "out of buffers"):
  with assert_checkpoints():
    with pytest.raises(OSError, match=match) as excinfo:
      ...
    ...
```
should work?

Reply:
By the way, it could be useful to have `@pytest.mark.parametrize` or `pytest-subtests` here so that separate checks would show up in the test report distinctly.

Reply:
Changed in f89cd0a. I didn't do the parameterize because it would be a lot of setup code re-running, but if this is important I can still change it.